text/0009-BundleTracker-integration.md

```diff
@@ -0,0 +1,588 @@
+
+In this API, the `BundleTracker` has a template argument `T`, which corresponds to the wrapper object type. This defaults to `Bundle` if no wrapper object is used. The `BundleTracker` constructor takes three arguments:
+
+- The `BundleContext` from which the tracker shall be created.
+- An unsigned 32-bit integer as a mask of states, where all of the active bits correspond to Bundle states that will be tracked. The `CreateStateMask` method is defined to simplify the creation of these masks
+- A shared pointer to a `BundleTrackerCustomizer` object. If `nullptr` is supplied to the constructor, the `BundleTracker` will use its internal `BundleTrackerCustomizer` methods.
+
+Once the `BundleTracker` is constructed, `Open()` is called to begin the tracking of bundles. When tracking is no longer necessary, `Close()` is called to stop tracking bundles. Additionally, `Close()` will be called automatically on an open `BundleTracker` if it goes out of scope.
+
+#### Using a BundleTracker
+
+Once a `BundleTracker` is opened, there are API methods to gain information and monitor bundles:
+
+- `GetBundles()` returns the `Bundle` objects currently being tracked.
+- `GetObject(Bundle)` returns the custom object paired to a `Bundle`.
+- `GetTracked()` returns all of the bundles and custom objects as an unordered map.
+- `IsEmpty()` tells the client whether bundles are being tracked.
+- `Size()` returns the number of currently tracked bundles.
+- `GetTrackingCount()` returns a tracking number that ticks up every time the `BundleTracker` changes, allowing for comparisons over time.
+
+#### Customizing a BundleTracker
+
+To implement the `BundleTrackerCustomizer` methods, there are two options that have the same effect:
+
+- Implementing a concrete subclass of `BundleTrackerCustomizer` and passing a `std::shared_ptr` to an instance into the `BundleTracker` constructor.
+- Subclassing `BundleTracker` and overriding the `BundleTrackerCustomizer` methods, then passing in a `nullptr` to the `BundleTracker` constructor.
+
+The `BundleTrackerCustomizer` methods offer the ability to create and manage custom objects to be tracked (the pointer returned from `AddingBundle(const Bundle&, const BundleEvent&)` will be passed into the other methods when called on the same `Bundle`). Additionally, if a `NULL` is returned, the object will not be tracked by the `BundleTracker`.
+
+The three callbacks are called under the following conditions:
+
+`AddingBundle`:
+
+- A `Bundle` enters a state covered by the `stateMask`, and is not currently being tracked.
+
+`RemovedBundle`:
+
+- A `Bundle` that is currently being tracked enters a state not covered by the `stateMask`.
+- A `Bundle` is removed manually using `Remove()`
+- A `Bundle` is currently being tracked when the `Close()` method is called on the `BundleTracker` (or when the `BundleTracker` goes out of scope).
+
+`ModifiedBundle`:
+
+- A `Bundle` enters a state covered by the `stateMask`, and is currently being tracked.
```

Comment:
This can be simplified by removing `, and is currently being tracked.` and starting the sentence with `A tracked Bundle`

Reply:
This sentence is simplified.